pkg/helm/controller.go

```diff
@@ -400,7 +400,7 @@ func args(chart *helmv1.HelmChart) []string {
 		if val.StrVal == "false" || val.StrVal == "true" {
 			args = append(args, "--set", fmt.Sprintf("%s=%s", k, val.StrVal))
 		} else if val.StrVal != "" {
-			args = append(args, "--set-string", fmt.Sprintf("%s=%s", k, val.StrVal))
+			args = append(args, "--set-string", fmt.Sprintf("%s=%s", k, strings.ReplaceAll(val.StrVal, ",", "\\,")))
```

Comment:
From the "Actual Behavior" at https://github.com/rancher/rke2/issues/948:
> parsing errors in the helm pods due to:
> --set-string 'global.clusterCIDR=[192.168.0.1/16' '2600:1f1c:ab4:ee48::/48]' should be:
> --set-string 'global.clusterCIDR=192.168.0.1/16,2600:1f1c:ab4:ee48::/48'

I fail to see how this addresses the apparent incorrect conversion from a string array to a comma-delimited string.

Reply:
This is answered in https://github.com/rancher/rke2/issues/948#issuecomment-832331358